build_projects/dotnet-host-build/CompileTargets.cs

```diff
@@ -171,7 +171,9 @@ public static BuildTargetResult CompileCoreHost(BuildTargetContext c)
         {
             var hostVersion = c.BuildContext.Get<HostVersion>("HostVersion");
             var configuration = c.BuildContext.Get<string>("Configuration");
-            string rid = c.BuildContext.Get<string>("TargetRID");
+            string rid = DotNetCli.Stage0.GetRuntimeId();
```

Comment:
I would suggest something like this that will preserve cross build semantics as well:

```
if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
{
...
    rid = $"win10-{platform.ToLower()}";
...
}
```

I would also suggest to rename the variable to "hostRID". And if you are doing this, I would suggest do this initialization in PrepareTargets.CommonInit and simply fetch it here from BuildContext.